crates/matrix-sdk/src/event_cache/mod.rs

```diff
@@ -501,6 +558,29 @@ impl RoomEventCache {
         None
     }
 
+    /// Try to find an event by id in this room, along with all relations.
+    pub async fn event_with_relations(&self, event_id: &EventId) -> Vec<SyncTimelineEvent> {
+        let relations = self.inner.relations.read().await;
+        let relation_ids = relations.get(event_id);
```

Comment:
Can you inline this where it's used, instead? It's not useful to preload it, if the event couldn't be found.